src/main/java/pl/edu/mimuw/Bank.java

```diff
@@ -1,25 +1,134 @@
 package pl.edu.mimuw;
 
-import java.util.List;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Calendar;
 
 public class Bank {
 
-  @Override
-  public String toString() {
-    throw new IllegalStateException("TODO: get information about bank and all its clients (use StringBuilder)");
-  }
+    private final ArrayList<BankClient> allClients = new ArrayList<>();
+    private final double depositInterestRate;
+    private final double loanInterestRate;
+    private Timestamp today = Timestamp.valueOf(java.time.LocalDateTime.now());
+    private int clientCount = 1;
+    private int actionCount = 1;
 
-  public List<BankClient> getAllClients() {
-    throw new IllegalStateException("TODO: remember not to return internal state list as it can be modified");
-  }
+    public Bank(double depositInterestRate, double loanInterestRate) {
+        this.depositInterestRate = depositInterestRate;
+        this.loanInterestRate = loanInterestRate;
+    }
 
-  public void addClient(BankClient client) {
-    throw new IllegalStateException("TODO");
-  }
+    private String assignActionNumber() {
+        return String.format("%04d", actionCount++);
+    }
 
-  public BankClient getClient(int clientIndex) {
-    final var clients = getAllClients();
-    if (clientIndex >= clients.size()) throw new IllegalArgumentException("Invalid client index");
-    return clients.get(clientIndex);
-  }
+    private String assignClientNumber() {
+        return String.format("%04d", clientCount++);
+    }
+
+    public void collectInstallments() {
+        for (var client : allClients)
+            for (var action : client.getActionHistory())
+                action.collect(this, client);
+    }
+
+    public void advanceCurrentDate(int days) {
+        Calendar date = Calendar.getInstance();
+
+        date.setTime(this.today);
+        date.add(Calendar.DATE, days);
+
+        this.today = Timestamp.from(date.toInstant());
+    }
+
+    public Timestamp getCurrentDate() {
+        return this.today;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder result = new StringBuilder();
+        result.append("BANK STATS\nList of clients:\n");
+
+        for (var client : allClients) {
+            result.append(client.getCredentials()).
```

Comment:
jednak działa kropka na końcu liniii xdd ale nie piszmy tak, piwerwszy raz to zobaczyłem, rozumiem że literówka :)